lib/configParser.ts

```diff
@@ -0,0 +1,258 @@
+import { resolve, dirname } from 'path';
+import { sync } from 'glob';
+import { Logger } from './logger';
+
+// Coffee is required here to enable config files written in coffee-script.
+try {
+  require('coffee-script').register();
+} catch (e) {
+  // Intentionally blank - ignore if coffee-script is not available.
+}
+
+// LiveScript is required here to enable config files written in LiveScript.
+try {
+  require('LiveScript');
+} catch (e) {
+  // Intentionally blank - ignore if LiveScript is not available.
+}
+
+export interface Config {
+  specs: Array<string>;
+  multiCapabilities: Array<any>;
+  rootElement: string;
+  allScriptsTimeout: number;
+  getPageTimeout: number;
+  params: any;
+  framework: string;
+  jasmineNodeOpts: {
+    showColors: boolean;
+    defaultTimeoutInterval: number;
+  };
+  seleniumArgs: Array<any>;
+  seleniumSessionId: string;
+  mochaOpts: {
+    ui: string;
+    reporter: string;
+  };
+  chromeDriver: string;
+  configDir: string;
+  plugins: Array<any>;
+  skipSourceMapSupport: boolean;
+  suite: string;
+  suites: any;
+}
+
+export class ConfigParser {
+  private config_: Config;
+  constructor() {
+    // Default configuration.
+    this.config_ = {
+      specs: [],
+      multiCapabilities: [],
+      rootElement: 'body',
+      allScriptsTimeout: 11000,
+      getPageTimeout: 10000,
+      params: {},
+      framework: 'jasmine',
+      jasmineNodeOpts: {
+        showColors: true,
+        defaultTimeoutInterval: (30 * 1000)
+      },
+      seleniumArgs: [],
+      seleniumSessionId: null,
+      mochaOpts: {
+        ui: 'bdd',
+        reporter: 'list'
+      },
+      chromeDriver: null,
+      configDir: './',
+      plugins: [],
+      skipSourceMapSupport: false,
+      suites: null,
+      suite: null
+    };
+
+  }
+
+  /**
+   * Resolve a list of file patterns into a list of individual file paths.
+   *
+   * @param {Array.<string> | string} patterns
```

Comment:
what's your long-term plan with the JSDoc type annotations? will these be kept in sync with the typescript typings? Is anything type-checking them? Does the docgen use them?

Seems to me these should just be removed in this change so later maintainers don't have to deal with them.